src/main/resources/com/google/api/codegen/go/main.snip

```diff
@@ -1,397 +1,209 @@
 @extends "go/header.snip"
 
-@snippet generateFilename(service)
-    {@context.getApiConfig.getPackageName}/{@context.getReducedServiceName(service)}_client.go
-@end
-
-@snippet generateClass(service, body)
+@snippet generate(view)
     {@headerComment()}
 
     // AUTO-GENERATED CODE. DO NOT EDIT.
 
-    package {@context.getPackageName}
-
-    {@importSection(service)}
+    package {@view.packageName}
 
-    {@body}
-@end
-
-@private importSection(service)
     import (
-    @join import : context.getImports(service)
-    {@import}
-    @end
+        "fmt"
+        "runtime"
+        "time"
+
+        "github.com/googleapis/gax-go"
+        "golang.org/x/net/context"
+        "google.golang.org/api/option"
+        "google.golang.org/api/transport"
+        "google.golang.org/grpc"
+        "google.golang.org/grpc/codes"
+        "google.golang.org/grpc/metadata"
     )
-@end
-
-@snippet generateBody(service)
-    {@globals(service)}
-
-    @let clientName = context.getClientName(service)
-        // {@clientName} is a client for interacting with {@service.getSimpleName}.
-        type {@clientName} struct {
-            // The connection to the service.
-            conn *grpc.ClientConn
-
-            // The gRPC API client.
-            client {@context.getServiceClientName(service)}
-
-            // The call options for this service.
-            CallOptions *{@context.getClientPrefix(service)}CallOptions
-
-    @end
-        // The metadata to be sent with each request.
-        metadata map[string][]string
-    }
-
-    {@constructor(service)}
-    {@pathTemplateMethods(service)}
-    {@methods(service)}
-
-    {@iterators(service)}
-@end
-
-@private globals(service)
-    @if context.getApiConfig.getInterfaceConfig(service).getCollectionConfigs
-        var (
-            @join collectionConfig : context.getApiConfig.getInterfaceConfig(service).getCollectionConfigs
-                {@pathTemplateVarName(service, collectionConfig)} = gax.MustCompilePathTemplate("{@collectionConfig.getNamePattern}")
-            @end
-        )
 
-    @end
+    var (
+        @join pathTemplate : view.pathTemplates
+            {@pathTemplate.name} = gax.MustCompilePathTemplate("{@pathTemplate.pattern}")
+        @end
+    )
 
-    // {@context.getClientPrefix(service)}CallOptions contains the retry settings for each method of this client.
-    type {@context.getClientPrefix(service)}CallOptions struct {
-        @join method : context.getNonStreamingMethods(service)
-            @let methodName = method.getSimpleName
-                {@methodName} []gax.CallOption
-            @end
+    // CallOptions contains the retry settings for each method of this client.
+    type {@view.name}CallOptions struct {
```

Comment:
Don't concatenate strings in the template, form them in the SurfaceNamer and pass them through the view model